api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java

```diff
@@ -264,4 +296,301 @@ public void saveConcept_shouldForceSetFlagIfSetMembersExist() throws Exception {
 		//Then
 		assertTrue(concept.getSet());
 	}
+	
+	/**
+	 * @see ConceptServiceImpl#retireConcept(Concept,String)
+	 * @Verifies should fail to retire concept if no reason is given
+	 */
+	@Test
+	public void retireConcept_shouldFailToRetireConceptIfReasonIsNotGiven() throws Exception {
+		Concept concept = conceptService.getConcept(3);
+		expectedException.expect(IllegalArgumentException.class);
+		
+		conceptService.retireConcept(concept,"");	
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#retireConcept(Concept,String)
+	 * @Verifies should retire concept
+	 */
+	@Test
+	public void retireConcept_shouldRetireConcept() throws Exception {
+		Concept concept = conceptService.getConcept(3);
+		conceptService.retireConcept(concept, "dummy reason for retirement");
+			
+		assertTrue(concept.isRetired());
+		assertEquals("dummy reason for retirement", concept.getRetireReason());
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getAllDrugs()
+	 * @Verifies should return all the drugs 
+	 */
+	@Test
+	public void getAllDrugs_shouldReturnAListOfAllDrugs() throws Exception {
+		List<Drug> allDrugs = conceptService.getAllDrugs(true);
+		//4 Drugs in the sample dataset
+		
+		assertEquals(4, allDrugs.size());
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getDrugs(String, Concept, boolean, boolean, boolean, Integer, Integer)
+	 * @Verifies return list of drugs
+	 */
+	@Test
+	public void getDrugs_shouldReturnListOfDrugs() throws Exception {
+		Concept concept = conceptService.getConceptByUuid("05ec820a-d297-44e3-be6e-698531d9dd3f");
+		List<Drug> drugs = conceptService.getDrugs("ASPIRIN", concept, true, true, true, 0, 100 );
+		assertEquals("ASPIRIN",drugs.get(0).getName());
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getDrug(String)
+	 * @Verifies should return the given Drug taking drugNumber or ID as a param
+	 */
+	@Test
+	public void getDrug_shouldReturnTheDrug() throws Exception {
+		Drug drug = conceptService.getDrugByUuid("05ec820a-d297-44e3-be6e-698531d9dd3f");
+		assertEquals(drug, conceptService.getDrug("ASPIRIN"));
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getDrug(String)
+	 * @Verifies should return null if no drug is found 
+	 */
+	@Test
+	public void getDrug_shouldReturnNullIfNoDrugIsFound() throws Exception {
+		assertNull(conceptService.getDrug(123456));
+				
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getAllConceptClasses(boolean)
+	 * @Verifies should return a list of all concept classes 
+	 */
+	@Test
+	public void getAllConceptClasses_shouldReturnAllConceptClasses() throws Exception {
+		List<ConceptClass> conceptClasses = conceptService.getAllConceptClasses(true);
+		assertEquals(20, conceptClasses.size());
+		
+		//check for not retired classes
```

Comment:
@mavrk remove comment. do not comment your code. in this case any dev can see in the IDE that the parameter false in getAllConceptClasses does what your comment says.